src/main/java/io/rtr/conduit/amqp/impl/AMQPConnection.java

```diff
@@ -0,0 +1,103 @@
+package io.rtr.conduit.amqp.impl;
+
+import com.rabbitmq.client.AlreadyClosedException;
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.ConnectionFactory;
+import com.rabbitmq.client.MetricsCollector;
+import io.rtr.conduit.amqp.transport.TransportExecutor;
+
+import javax.net.ssl.SSLSocketFactory;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Supplier;
+
+public class AMQPConnection {
+    private final ConnectionFactory connectionFactory;
+    private Connection connection;
+    private TransportExecutor executor;
+    private final Supplier<TransportExecutor> executorFactory;
+
+    public AMQPConnection(boolean ssl, String host, int port, MetricsCollector metricsCollector) {
+        this(new ConnectionFactory(), TransportExecutor::new, ssl, host, port, metricsCollector);
+    }
+
+    public AMQPConnection(ConnectionFactory factory, Supplier<TransportExecutor> executorFactory, boolean ssl, String host, int port, MetricsCollector metricsCollector) {
+        this.connectionFactory = factory;
+        this.executorFactory = executorFactory;
+        if (ssl) {
+            factory.setSocketFactory(SSLSocketFactory.getDefault());
+        }
+
+        factory.setHost(host);
+        factory.setPort(port);
+
+        if (metricsCollector != null) {
+            factory.setMetricsCollector(metricsCollector);
+        }
+    }
+
+    public boolean isConnected() {
+        return this.connection != null && this.connection.isOpen();
+    }
+
+    public synchronized void connect(AMQPConnectionProperties properties) throws IOException, TimeoutException {
+        if (isConnected()) {
+            return;
+        }
+        configureConnectionFactory(properties);
+        initializeExecutor();
+        connection = connectionFactory.newConnection(executor);
+    }
+
+    public synchronized void disconnect() throws IOException {
+        //! We are going to assume that closing an already closed
+        //  connection is considered success.
+        if (connection != null && connection.isOpen()) {
+            try {
+                connection.close(connectionFactory.getConnectionTimeout());
+            } catch (AlreadyClosedException ignored) {}
+        }
+        stopListening();
+    }
+
+    public Channel createChannel() throws IOException {
+        if (!isConnected()) {
+            throw new IllegalStateException("Attempted to create channel whilst disconnected.");
+        }
+        Channel channel = connection.createChannel();
+        channel.basicQos(1);
```

Comment:
What side effects does this have?

Reply:
It sets the client's message prefetch count to 1, basically don't prefetch - it results in more  network round trips but distributes messages more evenly across multiple instances.

The `basicQos` method call involves a synchronous network interaction, it's not just a client change

It's the same behaviour as prior to the PR, just moved from `AMQPTransport`